packages/ui/react-ui-editor/src/extensions/comments.ts

```diff
@@ -521,3 +546,23 @@ export const useComments = (view: EditorView | null | undefined, id: string, com
     }
   }, [id, view, comments]);
 };
+
+/**
+ * Hook provides an extension to compute the current comment state under the selection.
+ * NOTE(Zan): I think this conceptually belongs in 'formatting.ts' but we can't import ESM modules there atm.
```

Comment:
how is that ESM? i mean why are the two approaches done differently?

Reply:
Since formatting has an associated test file, and the tests for that project haven't been migrated to `vitest` we can't have any ESM imports in `formatting.ts`.